File: 2021/2021W01/PreppindataW01.py
import pandas as pd
import re

bikefix_pt = {'Mountain':'^M.*', 'Gravel':'^G.*', 'Road':'^R.*'}

df = pd.read_csv(r'F:/Data/PD 2021 Wk 1 Input - Bike Sales.csv')
df['Store'] = df['Store - Bike'].apply(lambda x: re.search('^(\w+)',x).group(1))
df['Bike'] = df['Store - Bike'].apply(lambda x: re.search('(\w+)$',x).group(1))

for correct, pattern in bikefix_pt.items():
    df['Bike'] = df['Bike'].replace(to_replace = pattern, value = correct, regex = True)

df['Date'] = df['Date'].apply(lambda x: pd.to_datetime(x,format='%d/%m/%Y').date())
df['Quarter'] = df['Date'].apply(lambda x: (x.month-1)//3+1)
df['Day of Month'] = df['Date'].apply(lambda x: x.day)

final = df[df['Order ID']>10][['Quarter', 'Day of Month', 'Store', 'Bike', 'Order ID', 'Customer Age', 'Bike Value', 'Existing Customer?']].copy()

File: 2021/2021W02/preppindataW01.R
library(dplyr)
library(tidyr)
library(readr)
library(stringr)
library(stringi)
library(lubridate)

final <- read_csv("F:/Data/PD 2021 Wk 1 Input - Bike Sales.csv", 
                  col_types = cols(`Date` = col_date(format = "%d/%m/%Y"))) %>%
  separate(`Store - Bike`,c('Store','Bike'),sep='\\s+-\\s+') %>%
  mutate('Bike' = stri_replace_all_regex(`Bike`, pattern = c('^M.*', '^G.*', '^R.*'), 
                                         replacement = c('Mountain', 'Gravel', 'Road'), vectorize_all = FALSE),
         'Quarter' = quarter(`Date`),
         'Day of Month' = day(`Date`)) %>%
  filter(`Order ID`>10) %>%
  select(c('Quarter', 'Day of Month', 'Store', 'Bike', 'Order ID', 'Customer Age', 'Bike Value', 'Existing Customer?'))

View(final)
